yellowbrick/utils/kneed.py

```
@@ -0,0 +1,208 @@
+        :param ymx_i: the normalized y value of a local maximum
+        """
+        return ymx_i - (self.S * np.diff(self.xsn).mean())
+
+    def find_knee(self, ):
+        """This function finds and returns the knee value, the normalized knee
+        value, and the x value where the knee is located.
+        :returns: tuple(knee, norm_knee, knee_x)
+        :rtype: (float, float, int)
+        )
+        """
+        if not self.xmx_idx.size:
+            warnings.warn("No local maxima found in the distance curve\n"
+                          "The line is probably not polynomial, try plotting\n"
+                          "the distance curve with plt.plot(knee.xd, knee.yd)\n"
+                          "Also check that you aren't mistakenly setting the curve argument", RuntimeWarning)
+            return None, None, None
+
+        mxmx_iter = np.arange(self.xmx_idx[0], len(self.xsn))
+        xmx_idx_iter = np.append(self.xmx_idx, len(self.xsn))
+
+        knee_, norm_knee_, knee_x = 0.0, 0.0, None
+        for mxmx_i, mxmx in enumerate(xmx_idx_iter):
+            # stopping criteria for exhasuting array
+            if mxmx_i == len(xmx_idx_iter) - 1:
+                break
+            # indices between maxima/minima
+            idxs = (mxmx_iter > xmx_idx_iter[mxmx_i]) * \
+                (mxmx_iter < xmx_idx_iter[mxmx_i + 1])
+            between_local_mx = mxmx_iter[np.where(idxs)]
+
+            for j in between_local_mx:
+                if j in self.xmn_idx:
+                    # reached a minima, x indices are unique
+                    # only need to check if j is a min
+                    if self.yd[j + 1] > self.yd[j]:
+                        self.Tmx[mxmx_i] = 0
+                        knee_x = None  # reset x where yd crossed Tmx
+                    elif self.yd[j + 1] <= self.yd[j]:
+                        warnings.warn("If this is a minima, "
+                                      "how would you ever get here:", RuntimeWarning)
```

Perhaps we should raise an exception here instead? If not, replacing with the `YellowbrickWarning` would also be appreciated (and a bit more detail about what's happening that's causing the warning).